html2/q2.js

```diff
@@ -0,0 +1,156 @@
+console.log(movies
+    .filter(x=>x.ActorName=='Sophia Williams').length);
+
+// 6. Insert an element
+// {
+//     "MovieName": "The Final Stage",
+//     "ActorName": "John Doe",
+//     "ReleaseDate": "2022-08-11"
+// }
+// as last element
+
+const entry={
+
+    "MovieName": "The Final Stage",
+    "ActorName": "John Doe",
+    "ReleaseDate": "2022-08-11"
+}
+
+movies.push(entry);
+movies.forEach(x => {
+    console.log(x)
+});
+
+//7. Check whether there exists any duplicate movie names present in the array
+const titles=movies.filter(x=>x.MovieName)
+const dup=new Set(titles);
+console.log(dup.size)
+console.log(movies.length)
+console.log(dup.size<movies.length?'Has duplicates':"no duplicates")
+
+//8. Create a new array starting from the movie "City of Shadows"
+
+//by keeping all elements same
+const rem=movies.filter(x=>x.MovieName != "City of Shadows" );
+const citymovie=movies.filter(x=>x.MovieName == "City of Shadows" );
+const newarray=[citymovie,...rem]
+console.log(newarray);
+
+//by slicing and starting from the given movie
+const startIndex = movies.findIndex(movie => movie.MovieName === "City of Shadows");
+const newMoviesArray = startIndex != -1 ? movies.slice(startIndex) : [];
+console.log(newMoviesArray);
+
+//9. List the distinct actor names in array
+console.log(movies.length)
+const actorlist=movies.map(x=>x.ActorName);
+const distinctactorlist=new Set(actorlist);
+distinctactorlist.forEach(x=>console.log(x));
+
+/*
+10. Insert an element
+		{
+   			 "MovieName": "Rich & Poor",
+    			"ActorName": "Johnie Walker",
+   	 		"ReleaseDate": "2023-08-11"
+ 		 }
+	as next element to movie “Love and Destiny”
+*/
+
+const newmovie=		
+{
+    "MovieName": "Rich & Poor",
+    "ActorName": "Johnie Walker",
+    "ReleaseDate": "2023-08-11"
+};
+movies.splice(movies.findIndex(x=>x.MovieName=='Love and Destiny')+1,0,newmovie);
+console.log(movies);
+
+//11. Display the count of distinct actor names in array
+const actorslist=movies.map(x=>x.ActorName);
+const distinctactorslist=new Set(actorslist);
+distinctactorlist.forEach(x=>console.log(x));
+console.log(distinctactorslist.size);
+
+//12. Remove the movie named  "The Last Stand"
+movies.splice(movies.findIndex(x=>x.MovieName=='The Last Stand'),1);
+console.log(movies);
+
+//13. Check whether all the movies are released after 2021 Dec 31
+console.log(movies.every(x=>x.ReleaseDate > '2021-12-31'));
+
+//14. Update movie named  "City of Shadows" ‘s release date as  "2023-03-13"
+for (let i = 0; i < movies.length; i++) {
+    if (movies[i].MovieName === "City of Shadows") {
+      movies[i].ReleaseDate = "2023-03-13";
+      break;
+    }
+  }
+  
+  console.log(movies);
+  
+//15. Create a new array of movie names whose movie name length is greater than 10.
+const nmovie=[];
+console.log(movies.length)
+for (let i = 0; i < movies.length; i++) {
+    if (movies[i].MovieName.length >10) {
+        nmovie.push(movies[i].MovieName)
+    }
+  }
```

Comment:
This will work fine,time complexity is O(n). in case we have a larger dataset possibly greater than 500, this is good
For smaller data sets use filter followed by map, which would be a single line of code if we are chaining the operators, this would be more readable